viewing_party/party.py

```diff
@@ -1,23 +1,241 @@
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    lod_watched.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    return user_data after adding movie to watchlist
+    An empty list represents that the user has no movies in their watchlist
+    user_data = {
+        'watchlist': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+    lod_watchlist = user_data["watchlist"]
+    lod_watchlist.append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    - if title is in watchlist: remove from watchlist, add to watched, return user data
+    - if movie is not in user watchlist, return user data
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}],
+        'watchlist': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watchlist = user_data["watchlist"]
+    lod_watched = user_data["watched"]
+    # returns [ {movie_dict} , {movie_dict} , ...]
+
+    for movie_dict in lod_watchlist:
+        if movie_dict["title"] == title:
+            # remove from watchlist
+            lod_watchlist.remove(movie_dict)
+            # add to watched
+            lod_watched.append(movie_dict)
+            # exit loop early if found
+            break
 
+    return user_data
+
+
+#wave 2 
+
+def get_watched_avg_rating(user_data):
+    '''
+    return the average rating
+    The average rating of an empty watched list is 0.0
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    total_rating = 0
+    
+    if lod_watched == []:
+        return 0.0
+    for movie in lod_watched:
+        total_rating += movie["rating"]
+    
+    avg_rating = total_rating/len(lod_watched)
+    return avg_rating
         
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+
+def get_most_watched_genre(user_data):
+    '''
+    return the genre that is the most frequently watched
+    If the value of "watched" is an empty list, get_most_watched_genre should return None.
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    genre_count = {}
+
+
+    if lod_watched == []:
+        return None
+    for movie in lod_watched:
+        genre = movie["genre"] # assigning value of genre to variable "genre"
+        count = genre_count.get(genre, 0)
+        genre_count[genre] = count + 1

```

Comment:
👍 Nice job using `get` to build up your frequency table. You'll commonly see lines 117-118 written like:
```python
genre_count[genre] = genre_count.get(genre, 0) + 1
```